apps/sparo-lib/src/services/GitSparseCheckoutService.ts

```diff
@@ -226,20 +225,30 @@ ${availableProfiles.join(',')}
 
     {
       const stopwatch: Stopwatch = Stopwatch.start();
-      if (checkoutAction === 'purge') {
-        // re-apply the initial paths for setting up sparse repo state
-        this._prepareMonorepoSkeleton({ restore: true });
-      }
-      // FIXME: too long CLI parameter
-      if (!(checkoutAction === 'purge' || checkoutAction === 'skeleton')) {
-        if (targetFolders.length === 0) {
-          terminal.writeDebugLine(`Skip sparse checkout regarding no target folders`);
-        } else {
-          terminal.writeLine(`Run sparse checkout for these folders: ${targetFolders.join(' ')}`);
-          this._sparseCheckoutPaths(targetFolders, {
-            action: 'add'
-          });
-        }
+      switch (checkoutAction) {
+        case 'purge':
+        case 'skeleton':
+          // re-apply the initial paths for setting up sparse repo state
+          this._prepareMonorepoSkeleton({ restore: checkoutAction === 'purge' });
+          break;
+        case 'add':
+        case 'set':
+          if (targetFolders.length === 0) {
+            terminal.writeDebugLine(`Skip sparse checkout regarding no target folders`);
+          } else {
+            terminal.writeLine(
+              `Run sparse checkout ${checkoutAction} for these folders: ${targetFolders.join(' ')}`
```

Comment:
```suggestion
              `Performing sparse checkout ${checkoutAction} for these folders: ${targetFolders.join(' ')}`
```

Reply:
This is a potentially long list of folders.  It should be `writeDebugLine()` instead of `writeLine()`, right?

If it's important to show the folders, then we should make sure they are formatted to be readable (e.g. `join('\n  ')`)

Reply:
I think error message should be shown to users. Let me change it with `'\n  '`